### PR TITLE
Merge bun changes

### PR DESCRIPTION
**Pull request by uNetworkingAB**

### CODE REVIEW

**Comment by Jarred-Sumner** [comment]
oh i didn't see this

cool

**Comment by uNetworkingAB** [comment]
Yeah I gradually take parts from here. But GitHub does not update its diff

File: benchmarks/load_test.c [hunk 1]
```diff
@@ -1,7 +1,29 @@
 /* This is a simple yet efficient WebSocket server benchmark much like WRK */
 
 #define _BSD_SOURCE
+
+#ifdef __APPLE__
+#include <libkern/OSByteOrder.h>
+
+#define htobe16(x) OSSwapHostToBigInt16(x)
```
Comment (Jarred-Sumner) [inline benchmarks/load_test.c:8]: rationale for this was to make `load_test` work on macOS

macOS is missing these functions

File: examples/EchoServer.cpp [hunk 2]
```diff
@@ -12,11 +12,17 @@ int main() {
 
     /* Keep in mind that uWS::SSLApp({options}) is the same as uWS::App() when compiled without SSL support.
      * You may swap to using uWS:App() if you don't need SSL */
-    uWS::SSLApp({
+    uWS::App({
```
Comment (Jarred-Sumner) [inline examples/EchoServer.cpp:15]: this was so I could run the EchoServer without HTTPS, but there was no need to really commit this

File: src/App.h [hunk 3]
```diff
@@ -153,6 +153,13 @@ struct TemplatedApp {
         httpContext->filter(std::move(filterHandler));
     }
 
+    /* Publishes a message to all websocket contexts - conceptually as if publishing to the one single
+     * TopicTree of this app (technically there are many TopicTrees, however the concept is that one
+     * app has one conceptual Topic tree) */
+    bool publish(std::string_view topic, std::string_view message, unsigned char opCode, bool compress = false) {
```
Comment (Jarred-Sumner) [inline src/App.h:159]: this is dumb and as discussed shouldn't include it

File: src/ChunkedEncoding.h [hunk 4]
```diff
@@ -29,12 +29,12 @@
 
 namespace uWS {
 
-    uint32_t STATE_HAS_SIZE = 0x80000000;
-    uint32_t STATE_IS_CHUNKED = 0x40000000;
-    uint32_t STATE_SIZE_MASK = 0x3FFFFFFF;
+    constexpr uint32_t STATE_HAS_SIZE = 0x80000000;
```
Comment (Jarred-Sumner) [inline src/ChunkedEncoding.h:32]: this was so we could include uWS in two different object files

In one, it's the C bindings

In the other, it's to copy the `Headers` object from JS bindings -> uWS response object

There were duplicate symbol errors without `constexpr` / `inline` / `static`

File: src/Loop.h [hunk 5]
```diff
@@ -79,6 +80,8 @@ struct Loop {
         /* We also need some timers (should live off the one 4 second timer rather) */
         LoopData *loopData = (LoopData *) us_loop_ext((struct us_loop_t *) loop);
         loopData->dateTimer = us_create_timer((struct us_loop_t *) loop, 1, sizeof(LoopData *));
+        loopData->updateDate();
```
Comment (Jarred-Sumner) [inline src/Loop.h:83]: This was to ensure we always have a date set

File: .gitmodules [hunk 6]
```diff
@@ -1,9 +1,12 @@
 [submodule "uSockets"]
-	path = uSockets
-	url = https://github.com/uNetworking/uSockets.git
+path = uSockets
+url = https://github.com/Jarred-Sumner/uSockets
```
Comment (Jarred-Sumner) [inline .gitmodules:3]: You probably don't want to commit this

File: src/Loop.h (hunk 5 above)
Comment (uNetworkingAB) [reply src/Loop.h:83]: It's set in the constructor

File: src/ChunkedEncoding.h (hunk 4 above)
Comment (uNetworkingAB) [reply src/ChunkedEncoding.h:32]: this whole part is already included (but github diff doesn't update)

File: src/App.h (hunk 3 above)
Comment (uNetworkingAB) [reply src/App.h:159]: right :)

File: examples/EchoServer.cpp (hunk 2 above)
Comment (uNetworkingAB) [reply examples/EchoServer.cpp:15]: yep, examples won't change

File: benchmarks/load_test.c (hunk 1 above)
Comment (uNetworkingAB) [reply benchmarks/load_test.c:8]: already included

File: .gitmodules (hunk 6 above)
Comment (uNetworkingAB) [reply .gitmodules:3]: you can checkout your variant until uSockets is merged

**Review by N0N1m3 — APPROVED** [review]

**Pull request closed without merging**